Tests/Client/Unit Tests/Client+DevicesTests.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  Client+DevicesTests.swift
+//  StreamChatClientTests
+//
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import XCTest
+
+@testable import StreamChatClient
+
+class Client_DevicesTests: XCTestCase {
+
+    // Ideally, we would create a new client for every test case. Unfortunately, this is not technically
+    // possible at the moment.
+    lazy var client: Client = {
+        let sessionConfig = URLSessionConfiguration.default
+
+        sessionConfig.protocolClasses?.insert(RequestRecorderURLProtocol.self, at: 0)
+        Client.config = .init(apiKey: "test_api_key", defaultURLSessionConfiguration: sessionConfig)
+        return Client.shared
+    }()
+
+    func test_getDevice_createsRequest() {
+        // Setup
+        let testUser = User(id: "test_user_\(UUID())")
+        client.set(user: testUser, token: "test_token")
+
+        // Action
+        client.devices { _ in }
+
+        // Assert
+        AssertNetworkRequest(
+            method: .get,
+            path: "/devices",
+            headers: ["Content-Type": "application/json"],
+            queryParameters: ["api_key": "test_api_key"],
+            body: nil
+        )
+    }
+
+    func test_addDevice_withDeviceID_createsRequest() {
+        // Setup
+        let testUser = User(id: "test_user_\(UUID())")
+        let testDeviceId = "device_id_\(UUID())"
+        client.set(user: testUser, token: "test_token")
```

Comment:
These shared lines could be moved into `setUp`, or event in the computed property itself

Reply:
I thought about it but didn't want to overcomplicate things. I think we'll find the right pattern for this while adding more tests. But good comment 👍

Reply:
Isn't keeping setup part individual for each test (although they're the same) overcomplicating?

Reply:
I like how local everything is. You can copy-paste the function somewhere else and it will still work (mostly). I like it when tests are dumb and straightforward. When you don't need to think about the code that much and it's just there right in your face.

This is just my personal preference and I have no best-practices link I can post here. If it bothers you, feel free to add a commit and show your preferred way of doing this 👍

Reply:
I see, it's a tradeoff between locality/ease of understanding vs less repetition. Let's keep it as it is for this PR 👍🏼
Disclosure: when I write tests, I'll use `setUp` function 😂

Reply:
I refactored the lines shared by all tests cases to the `setUp` function